src/wp-includes/blocks/index.php

```diff
@@ -40,26 +39,60 @@ function register_core_block_style_handles() {
 		'editorStyle' => 'editor',
 	);
 
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require $includes_path . 'blocks/blocks-json.php';
+	}
+
+	$files          = false;
+	$transient_name = 'wp_core_block_css_files';
+
 	/*
 	 * Ignore transient cache when the development mode is set to 'core'. Why? To avoid interfering with
 	 * the core developer's workflow.
 	 */
-	if ( ! wp_is_development_mode( 'core' ) ) {
-		$transient_name = 'wp_core_block_css_files';
-		$files          = get_transient( $transient_name );
-		if ( ! $files ) {
-			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
-			set_transient( $transient_name, $files );
+	$can_use_cached = ! wp_is_development_mode( 'core' );
+
+	if ( $can_use_cached ) {
+		$cached_files = get_transient( $transient_name );
+
+		// Check the validity of cached values by checking against the current WordPress version.
+		if (
+			is_array( $cached_files )
+			&& isset( $cached_files['version'] )
+			&& $cached_files['version'] === $wp_version
+			&& isset( $cached_files['files'] )
+		) {
+			$files = $cached_files['files'];
 		}
-	} else {
+	}
+
+	if ( ! $files ) {
 		$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
+		$files = array_map(
+			static function ( $file ) use ( $includes_path ) {
+				return str_replace( $includes_path, '', $file );
+			},
+			$files
+		);
```

Comment:
```suggestion
		$raw_files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
		$files = array();
		foreach( $raw_files as  $file ){
		   $files[] = str_replace( $includes_path, '', $file );
		}
```
This feels a little more readable.

Reply:
I don't feel strongly about this, but `array_map` seems totally fine to me. It's used already in several places in core and is a syntax that is fairly common in the broader PHP ecosystem.